builder/prepare-ubuntu.sh

```diff
@@ -7,9 +7,21 @@
 
 # variables
 
-[ -z "${GO_VERSION}" ] && GO_VERSION="1.21"
-[ -z "${CLANG_VERSION}" ] && CLANG_VERSION="12"
-[ -z "${ARCH}" ] && ARCH="amd64"
+[ -z "${GO_VERSION}" ] && GO_VERSION="1.22"
+[ -z "${CLANG_VERSION}" ] && CLANG_VERSION="14"
+[ -z "${ARCH}" ] && ARCH=$(uname -m)
+
+case "${ARCH}" in
+    x86_64|amd64)
+        ARCH="amd64"
+        ;;
+    aarch64|arm64)
+        ARCH="arm64"
+        ;;
+    *)
+        die "unsupported architecture ${ARCH}"
+        ;;
+esac
```

Comment:
ARCH is not currently used but might be if we need to download binaries instead of installing them via the package system.